engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/NumberPatternValidator.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.constraintvalidators.hv;
+
+import java.math.BigDecimal;
+import java.text.DecimalFormat;
+import java.util.regex.Matcher;
+import java.util.regex.PatternSyntaxException;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.constraints.Pattern;
+
+import org.hibernate.validator.constraints.NumberPattern;
+import org.hibernate.validator.internal.util.logging.Log;
+import org.hibernate.validator.internal.util.logging.LoggerFactory;
+
+/**
+ * Checks that given {@link Number} matches provided regexp.
+ *
+ * @author Marko Bekhta
+ */
+public class NumberPatternValidator implements ConstraintValidator<NumberPattern, Number> {
+
+	private static final Log log = LoggerFactory.make();
+
+	private java.util.regex.Pattern pattern;
```

Comment:
Once you have removed the dependency to Pattern.Flag, you'll be able to remove the fqcn.